server/src/main/java/org/elasticsearch/index/engine/Engine.java

```diff
@@ -177,6 +179,23 @@ public DocsStats docStats() {
         }
     }
 
+    /**
+     * Returns the file sizes for the current commit
+     */
+    public List<SegmentFileSize> getLastCommittedSegmentFileSizes() {
```

Comment:
Would it make more sense to return a `Map<String, Long>`, then you don't need to define a separate type (and also protect against dups)?

Reply:
Changed.